library/std/src/sys/unix/ext/io.rs

```diff
@@ -25,6 +25,18 @@ pub trait AsRawFd {
     /// This method does **not** pass ownership of the raw file descriptor
     /// to the caller. The descriptor is only guaranteed to be valid while
     /// the original object has not yet been destroyed.
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// use std::fs::File;
+    /// use std::os::unix::io::AsRawFd;
+    ///
+    /// fn main() -> std::io::Result<()> {
+    ///     let mut f = File::open("foo.txt")?;
+    ///     let raw_fd = f.as_raw_fd();
+    /// }
+    /// ```
```

Comment:
Perhaps the example should also highlight (as the prose already says) that the `RawFd` becomes invalid once `f` gets dropped.

Reply:
I added a comment just above. :)

Reply:
What happens if it becomes invalid, should we show an example of how to use it incorrectly? But I think this should be encoded in the type system if possible but that would probably be a breaking change, the comment by itself does not do much.

I tried and looks like it still compiles, potential footgun maybe?

```rust
use std::fs::File;
use std::os::unix::io::{AsRawFd, RawFd};

fn main() -> std::io::Result<()> {
    let mut f = File::open("Cargo.toml")?;
    // Note that `raw_fd` is only valid as long as `f` exists.
    let raw_fd: RawFd = f.as_raw_fd();
    dbg!(raw_fd);
    drop(f);
    dbg!(raw_fd); // raw_fd still available which I think it shouldn't be
    todo!();
    Ok(())
}
```

Reply:
`raw_fd` is just an integer, but the ID itself isn't valid if try any I/O operation on it. That's why it's very unsafe to use the `raw_fd` methods.